core/src/test/java/io/confluent/rest/SslTest.java

```diff
@@ -0,0 +1,327 @@
+      File untrustedClient = File.createTempFile("SslTest-client-keystore", ".jks");
+      Map<String, X509Certificate> certs = new HashMap<>();
+      createKeystoreWithCert(untrustedClient, "client", certs);
+
+      try {
+        makeGetRequestAndVerify200("https://localhost:" + (RestConfig.PORT_HTTPS_CONFIG_DEFAULT) + "/test",
+                untrustedClient.getAbsolutePath(), SSL_PASSWORD, SSL_PASSWORD);
+      } catch (SSLHandshakeException she) {
+        // expected.
+      } catch (SocketException se) {
+        // expected. SocketException is thrown when SSLHandshakeException is caught.
+      }
+    } finally {
+      app.stop();
+    }
+  }
+
+  @Test
+  public void testHttpsWithAuthAndNoClientCert() throws Exception {
+    Properties props = new Properties();
+    props.put(RestConfig.REST_PROTOCOL_CONFIG, RestConfig.REST_PROTOCOL_HTTPS);
+    configServerKeystore(props);
+    configServerTruststore(props);
+    enableSslClientAuth(props);
+    TestRestConfig config = new TestRestConfig(props);
+    SslTestApplication app = new SslTestApplication(config);
+    try {
+      app.start();
+
+      try {
+        makeGetRequestAndVerify200("https://localhost:" + (RestConfig.PORT_HTTPS_CONFIG_DEFAULT) + "/test");
+      } catch (SSLHandshakeException she) {
+        // expected.
+      } catch (SocketException se) {
+        // expected. SocketException is thrown when SSLHandshakeException is caught.
+      }
+    } finally {
+      app.stop();
+    }
+  }
+
+  private void makeGetRequestAndVerify200(String url, String clientKeystoreLocation, String clientKeystorePassword,
+                                          String clientKeyPassword)
+      throws Exception {
+    log.debug("Making GET " + url);
+    HttpGet httpget = new HttpGet(url);
+    CloseableHttpClient httpclient;
+    if (url.startsWith("http://")) {
+      httpclient = HttpClients.createDefault();
+    } else {
+      // trust all self-signed certs.
+      SSLContextBuilder sslContextBuilder = SSLContexts.custom()
+              .loadTrustMaterial(new TrustSelfSignedStrategy());
+
+      // add the client keystore if it's configured.
+      if (clientKeystoreLocation != null) {
+        sslContextBuilder.loadKeyMaterial(new File(clientKeystoreLocation),
+                clientKeystorePassword.toCharArray(),
+                clientKeyPassword.toCharArray());
+      }
+      SSLContext sslContext = sslContextBuilder.build();
+
+      SSLConnectionSocketFactory sslSf = new SSLConnectionSocketFactory(sslContext, new String[]{"TLSv1"},
+              null, SSLConnectionSocketFactory.getDefaultHostnameVerifier());
+
+      httpclient = HttpClients.custom()
+              .setSSLSocketFactory(sslSf)
+              .build();
+    }
+
+    CloseableHttpResponse response = httpclient.execute(httpget);
+    assertEquals("Response status must be 200.", 200, response.getStatusLine().getStatusCode());
+    response.close();
+    httpclient.close();
```

Comment:
You probably want to have a finally block to ensure `response` and `httpclient` are closed if an exception is thrown